### PR TITLE
fix: Add truth particle selection for truth tracking

### PR DESCRIPTION
**Pull request by XiaocongAi**
This PR adds the truth particle pre-selection, e.g. selecting only truth particles that have non-zero number of truth hits. This is important for tracking efficiency validation where particle with no hits on detector should not be taken into account.

### CODE REVIEW

**Review by msmk0 — APPROVED** [review]

**Comment by msmk0** [comment]
@XiaocongAi Could you update the branch to master.

File: Examples/Run/Reconstruction/RecTruthTracks.cpp [hunk 1]
```diff
@@ -21,6 +21,7 @@
 #include "ActsExamples/Options/CommonOptions.hpp"
 #include "ActsExamples/Plugins/BField/BFieldOptions.hpp"
 #include "ActsExamples/TruthTracking/ParticleSmearing.hpp"
+#include "ActsExamples/TruthTracking/TruthSeedSelector.hpp"
```
Comment (asalzburger) [inline Examples/Run/Reconstruction/RecTruthTracks.cpp:24]: Did this `TruthSeedSelector` already exist?

**Review by asalzburger — APPROVED** [review]
Moritz already approved, I just had a question for clarification.

**Comment by asalzburger** [comment]
I've updated the branch on the web-interface, waiting for CI to finish and we can merge in.

File: Examples/Run/Reconstruction/RecTruthTracks.cpp (hunk 1 above)
Comment (XiaocongAi) [reply Examples/Run/Reconstruction/RecTruthTracks.cpp:24]: Yes, it's already used for CKF.

**Comment by XiaocongAi** [comment]
Sorry, wrong forced push by mistake, which triggers the CI again...

**Pull request merged**